src/Particle/VirtualParticleSet.h

```diff
@@ -95,6 +95,15 @@ class VirtualParticleSet : public ParticleSet
                  bool sphere = false,
                  int iat     = -1);
 
+  inline size_t getTotalNum() const { return TotalNum; }
+  /**Extract list of Distance Tables
+    */
+  static const RefVectorWithLeader<const DistanceTableAB> extractDTRefList_vp(
+      const RefVectorWithLeader<const VirtualParticleSet>& vp_list,
+      int id);
+
+  static const std::vector<QMCTraits::PosType> extractCoordsRefList_vp(
```

Comment:
Please document this function.

Reply:
done

Reply:
This function doesn't make sense at all. VP doesn't have active particle coordinates. Something is wrong in the caller side.

Reply:
I updated it and gave it a better name

```
const std::vector<QMCTraits::PosType> VirtualParticleSet::extractVPCoords(
    const RefVectorWithLeader<const VirtualParticleSet>& vp_list)
{
  std::vector<QMCTraits::PosType> coords_list;
  for (const VirtualParticleSet& vp : vp_list)
    for (int iat = 0; iat < vp.getTotalNum(); iat++)
      coords_list.push_back(vp.R[iat]);

  return coords_list;
}
```

I'm not sure the returned vector needs to be const, and I don't think std::vector<PosType> is a great choice, but this part needs to be cleaned up anyway.

I think the ideal case would be to just directly access the coordinates through the distance table mw_mem, but I'm not sure how to do that.
Could I get the DistanceTableAB from the leader and then get [`mw_r_dr`](https://github.com/kgasperich/qmcpack/blob/65a2d8ffd5b9cd3b055750f8770c9ab2e18e0295/src/Particle/SoaDistanceTableABOMPTarget.h#L46) from that distance table? (after casting it to a `SoaDistanceTableABOMPTarget`?)